### ISSUE TITLE
In some circumstances, adding a gadget to an empty placeholder does not show configuration options

### ISSUE THREAD

**Issue by mfisher11** [Rise-Vision/gadgets#2]
Steps to reproduce:
0. This should be done in incognito or with a clear cache.
1. Go to risevision.com
2. Log in.
3. From the app launcher, choose editor.
4. Open/create a new presentation.
5. Add an item/placeholder.
6. Add one of these gadgets:  rss, headline news, financial chart (possibly others)

Result:
The modal contains an empty space where the configuration options should be and there is a gray shield (mixed content warning) at the right side of the address bar.

How it should work:
Configuration options should appear at the bottom of the modal.

Link to Presentation:
N/A

OS:
Linux, Windows

Chrome Ver:
44.0.2403.89

**Comment by mfisher11** [comment]
@donnapep Some additional information discovered as a result of community post (https://community.risevision.com/rise_vision_inc/topics/inability-to-edit-gadgets) and research:

This is caused by one or more references to insecure content when accessing rva over an encrypted connection.  If the user has opened http://rvauser.appspot.com/ or http://rva.risevision.com directly, the problem does not occur.  If they access it from www.risevision.com, it will automatically redirect to https and then any linked navigation to rva will also be over an https connection.
This is not a problem with any of our widgets because the UI URL has been updated to use https, however, the URL string doesn't seem to matter on the gadgets.  The three listed in the bug report are all configured for https and still have the problem.

The screenshot included in the community post shows a spreadsheet gadget. I created copies of the two deprecated gadgets and was able to replicate his issue with the one that DID specify https (Google Spreadsheet (Deprecated)), while the other one (Rise Vision Google Spreadsheet (OLD)) specified http in the URL and worked fine.

My theory is that there a reference inside the widget code that is hardcoded with http.
Is that the case?  Would simply changing that (or those) to https fix the problem?

Thanks.

**Comment by donnapep** [comment]
@mfisher11 Yes, in Gadgets we were specifying http or https explicitly. In Widgets we started using relative URLS and left out the protocol altogether, which is likely why they are not a problem.

**Comment by donnapep** [comment]
@mfisher11 Here's a version of the Google Spreadsheet Gadget that uses relative URLs that you can use to test against and see how it goes. @alanclayton FYI.

**Comment by mfisher11** [comment]
@donnapep That worked just fine, both in initial opening, preview and reconfiguration over https.  Thanks!

**Comment by alanclayton** [comment]
@mfisher11 what about on a running Display?

**Comment by mfisher11** [comment]
@alanclayton Yes.  It runs fine on a display as well.

**Comment by alanclayton** [comment]
@stulees can you take deploying this Gadget since @donnapep is out. And can you coordinate that deployment with Support so that they can smoke test as soon as you deploy? @mfisher11 @AlexKolenoff @btfrise Who is free to smoke test this?

**Comment by stulees** [comment]
@alanclayton I won't have @donnapep to review the PRs I submit for all the Gadget work today. It's important she reviews because this is the first time I've ever touched Gadgets. I think it will be best to have this all ready for her to review Monday and then she can deploy (I won't be working Monday).

**Comment by alanclayton** [comment]
@stulees okay

**Comment by robb-price** [comment]
@mfisher11 Updated the list of Gadgets this occurs on in the bug description.
